microsoft/testsuites/xfstests/xfstests.py

```diff
@@ -479,15 +721,163 @@ def save_xfstests_log(
+        test_status: The test status for the given test case
+        Returns:
+        The method returns the stack info message for the given test case
+        Example:
+        xfstest.create_xfstest_stack_info(
+            case="generic/001",
+            test_section="xfs",
+            test_status="FAILED"
+        )
+        Note: When running LISA in debug mode, expect verbose messages from 'ls' tool.
+        This is because the method checks for file existence per case in the results
+        dir.
+        This is normal behavior and can be ignored. We are working on reducing verbosity
+        of 'ls' calls to improve performance.
+        """
+
+        # Get XFSTest current path. we are looking at results/{test_type} directory here
+        xfstests_path = self.get_xfstests_path()
+        test_class = case.split("/")[0]
+        test_id = case.split("/")[1]
+        result_path = xfstests_path / f"results/{test_section}/{test_class}"
+        return_message: str = ""
+        # <TODO> this needs to be fixed as it's spilling over to console output.
+        if self.node.tools[Ls].path_exists(str(result_path), sudo=True):
+            # Note. This will dump a lot of output on debug console screen.
+            # Only un-comment for debugging.
+            # self._log.debug(
+            #     f"Found files in path {result_path} : "
+            #     f"{self.node.tools[Ls].list(str(result_path), sudo=True)}"
+            # )
+            # If passed, we only need DMESG output
+            if test_status == "PASSED":
+                dmesg_result = self.node.tools[Cat].run(
+                    f"{result_path}/{test_id}.dmesg", force_run=True, sudo=True
+                )
+                return_message = f"DMESG: {dmesg_result.stdout}"
+            # If failed, we need dmesg with diff output
+            elif test_status == "FAILED":
+                dmesg_result = self.node.tools[Cat].run(
+                    f"{result_path}/{test_id}.dmesg", force_run=True, sudo=True
+                )
+                full_out = result_path / f"{test_id}.full"
+                fail_out = result_path / f"{test_id}.out.bad"
+                # check of "full_out" and "fail_out" file exists
+                # Only then call diff tool.
+                if self.node.tools[Ls].path_exists(
+                    str(full_out), sudo=True
+                ) and self.node.tools[Ls].path_exists(str(fail_out), sudo=True):
+                    diff_result = self.node.tools[Diff].comparefiles(
+                        src=full_out,
+                        dest=fail_out,
+                    )
+                # else if full_out is null, return the fail_out file content.
+                # In some test cases, full_out is not generated due to permissions
+                # or other issues. However a fail file will always exists in such cases.
+                elif self.node.tools[Ls].path_exists(str(fail_out), sudo=True):
```

Comment:
There is a case I saw there is no fail out but full out, then below exception was thrown out.
```
  File "C:\app\lsg-lisa\lisa\microsoft\testsuites\xfstests\xfstests.py", line 869, in create_xfstest_stack_info
    return_message = f"DIFF: {diff_result}\n\nDMESG: {dmesg_result}"
                              ^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'diff_result' where it is not associated with a value
```